Sprint_6

Freestyle95 opened the pull request.

Commit:
gitignore

Commit:
DI

Commit:
implementation

Commit:
implementation

Commit:
Merge remote-tracking branch 'origin/develop' into develop

nikolay-977 commented on src/main/java/com/example/Lion.java:
```diff
@@ -6,17 +6,18 @@ public class Lion {
 
     boolean hasMane;
```
Необходимо исправить: для инстанс переменных следует использовать модификатор доступ private, либо protected, а взаимодействовать с переменными через геттеры и сеттеры.

nikolay-977 commented on src/main/java/com/example/Lion.java:
```diff
@@ -6,17 +6,18 @@ public class Lion {
     }
 
-    Feline feline = new Feline();
+    IFeline feline;
```
Необходимо исправить: В начале класса следует описывать переменные, а затем конструкторы и методы. Также можно установить курсор на корневую директорию и нажать сочетание клавиш Ctrl+Alt+L (Option+Command+L для Mac), затем выбрать checkbox rearrange entries и нажать ok.

nikolay-977 commented on src/test/java/com/example/AlexFromMadagascarTest.java:
```diff
@@ -0,0 +1,37 @@
+
+    @Test
+    public void getPlaceOfLiving() {
+        assertEquals("Нью-Йоркский зоопарк",alex.getPlaceOfLiving());
```
Необходимо исправить: для расстановки пробелов и табуляции в соответствии с настройками Code Style в Idea используй сочетание клавиш Ctrl+Alt+L. Также можно установить курсор на корневую директорию и нажать клавиш Ctrl+Alt+L, отметить чек-боксы: cleanup code, а также optimize imports(для оптимизации импортов) и rearrange entries(для изменения порядка переменных и методов в соответствии с best practics)

nikolay-977 commented on src/test/java/com/example/AnimalParameterizedTest.java:
```diff
@@ -0,0 +1,35 @@
+        this.foodListExpected = foodList;
+    }
+
+    @Parameterized.Parameters // добавили аннотацию
```
Можно улучшить: В параметризованных тестах для аннотации `@Parameterized.Parameters` лучше использовать аргумент name: `@Parameterized.Parameters(name = "Тестовые данные: {0} {1}")`, где {0}, {1} - индексы параметров. Это повысит информативность проверки.

nikolay-977 commented on src/test/java/com/example/LionParameterizedTest.java:
```diff
@@ -0,0 +1,55 @@
+        this.sex = sex;
+        this.hasManeExpected = hasManeExpected;
+    }
+
```
Необходимо исправить: убери лишние пропуски строк. Для зрительного разделения кода достаточно использовать один пропуск строки. Пропуски строк должны быть однообразными.

Commit:
Fixes due to review comments:
- Fixed code style
- Fixed access modifiers, added setter/getter
- Added assertion fail messages
- Added parameter names for parameterized tests

Freestyle95 commented:
@nikolay-977  замечания исправлены
